src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingExerciseResource.java

```diff
@@ -313,6 +348,10 @@ public ProgrammingExerciseResource(ProgrammingExerciseRepository programmingExer
         try {
             // Setup all repositories etc
             ProgrammingExercise newProgrammingExercise = programmingExerciseService.createProgrammingExercise(programmingExercise);
+            // Create default static code analysis categories
+            if (Boolean.TRUE.equals(programmingExercise.isStaticCodeAnalysisEnabled())) {
+                staticCodeAnalysisService.createDefaultCategories(programmingExercise);
```

Comment:
```suggestion
                staticCodeAnalysisService.createDefaultCategories(newProgrammingExercise);
```
Use the correct exercise here.